src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,469 @@
+
+        Returns:
+            pixel_values: torch.FloatTensor of shape (batch_size, 1, height, width)
+
+        """
+        return pixel_values[:, 0, :, :][:, None, :, :]
+
+
+SUPERPOINT_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`SuperPointConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+    """
+
+SUPERPOINT_INPUTS_DOCSTRING = r"""
+Args:
+    pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+        Pixel values. Pixel values can be obtained using [`SuperPointImageProcessor`]. See
+        [`SuperPointImageProcessor.__call__`] for details.
+    output_hidden_states (`bool`, *optional*):
+        Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for more
+        detail.
+    return_dict (`bool`, *optional*):
+        Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+    """
+
+
+@add_start_docstrings(
+    "SuperPoint model outputting keypoints and descriptors.",
+    SUPERPOINT_START_DOCSTRING,
+)
+class SuperPointModel(SuperPointPreTrainedModel):
+    """
+    SuperPoint model. It consists of a SuperPointEncoder, a SuperPointInterestPointDecoder and a
+    SuperPointDescriptorDecoder. SuperPoint was proposed in `SuperPoint: Self-Supervised Interest Point Detection and
+    Description <https://arxiv.org/abs/1712.07629>`__ by Daniel DeTone, Tomasz Malisiewicz, and Andrew Rabinovich. It
+    is a fully convolutional neural network that extracts keypoints and descriptors from an image. It is trained in a
+    self-supervised manner, using a combination of a photometric loss and a loss based on the homographic adaptation of
+    keypoints. It is made of a convolutional encoder and two decoders: one for keypoints and one for descriptors.
+    """
+
+    def __init__(self, config: SuperPointConfig) -> None:
+        super().__init__(config)
+
+        self.config = config
+
+        self.encoder = SuperPointEncoder(config)
+        self.keypoint_decoder = SuperPointInterestPointDecoder(config)
+        self.descriptor_decoder = SuperPointDescriptorDecoder(config)
+
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(SUPERPOINT_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=ImagePointDescriptionOutput,
+        config_class=_CONFIG_FOR_DOC,
+        modality="vision",
+    )
+    def forward(
```

Comment:
By convention, the model should still accept `labels` as an input. We can raise an exception immediately in the forward pass if it's no `None` saying training isn't possible with this model

Reply:
Added this in [this commit](https://github.com/huggingface/transformers/pull/28966/commits/d36c58e0b9ae77703e876b5bbf72ec98df90c444). I added tests to ensure we raise the error when labels are indeed not None